src/3d/qgsshadowrenderingframegraph.cpp

```diff
@@ -133,11 +133,54 @@ Qt3DRender::QFrameGraphNode *QgsShadowRenderingFrameGraph::constructPostprocessi
   mPostprocessClearBuffers = new Qt3DRender::QClearBuffers( mPostprocessPassLayerFilter );
   mPostprocessClearBuffers->setClearColor( QColor::fromRgbF( 0.0f, 0.0f, 0.0f ) );
 
+  //
+  {
+    mRenderCaptureTargetSelector = new Qt3DRender::QRenderTargetSelector( mPostprocessPassLayerFilter );
+
+    Qt3DRender::QRenderTarget *renderTarget = new Qt3DRender::QRenderTarget( mRenderCaptureTargetSelector );
+
+    // The lifetime of the objects created here is managed
+    // automatically, as they become children of this object.
+
+    // Create a render target output for rendering color.
+    Qt3DRender::QRenderTargetOutput *colorOutput = new Qt3DRender::QRenderTargetOutput( renderTarget );
+    colorOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Color0 );
+
+    // Create a texture to render into.
+    mRenderCaptureColorTexture = new Qt3DRender::QTexture2D( colorOutput );
+    mRenderCaptureColorTexture->setSize( mSize.width(), mSize.height() );
+    mRenderCaptureColorTexture->setFormat( Qt3DRender::QAbstractTexture::RGB8_UNorm );
+    mRenderCaptureColorTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+    mRenderCaptureColorTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+
+    // Hook the texture up to our output, and the output up to this object.
+    colorOutput->setTexture( mRenderCaptureColorTexture );
+    renderTarget->addOutput( colorOutput );
+
+    Qt3DRender::QRenderTargetOutput *depthOutput = new Qt3DRender::QRenderTargetOutput( renderTarget );
+
+    depthOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Depth );
+    mRenderCaptureDepthTexture = new Qt3DRender::QTexture2D( depthOutput );
+    mRenderCaptureDepthTexture->setSize( mSize.width(), mSize.height() );
+    mRenderCaptureDepthTexture->setFormat( Qt3DRender::QAbstractTexture::DepthFormat );
+    mRenderCaptureDepthTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+    mRenderCaptureDepthTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+    mRenderCaptureDepthTexture->setComparisonFunction( Qt3DRender::QAbstractTexture::CompareLessEqual );
+    mRenderCaptureDepthTexture->setComparisonMode( Qt3DRender::QAbstractTexture::CompareRefToTexture );
+
+    depthOutput->setTexture( mRenderCaptureDepthTexture );
+    renderTarget->addOutput( depthOutput );
+
+    mRenderCaptureTargetSelector->setTarget( renderTarget );
+  }
+  //
+
+  mRenderCapture = new Qt3DRender::QRenderCapture( mRenderCaptureTargetSelector );
```

Comment:
isn't the render capture and its textures eat some of memory resources even if we don't need to capture frames? Or is that negligible? (I am wondering if it is not better to attach the textures and render capture only when we need them - but if that adds complexity, probably not worth it)

Reply:
I think you are right, I will try that and compare the frame rate

Reply:
Frame rate does actually increase when I disable the render capture subtree. I added a way to toggle that part on/off.